vms/proposervm/block/build.go

```diff
@@ -6,39 +6,140 @@ package block
 import (
 	"crypto"
 	"crypto/rand"
+	"encoding/binary"
 	"time"
 
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/staking"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
 	"github.com/ava-labs/avalanchego/utils/hashing"
-	"github.com/ava-labs/avalanchego/utils/wrappers"
 )
 
 func BuildUnsigned(
 	parentID ids.ID,
 	timestamp time.Time,
 	pChainHeight uint64,
 	blockBytes []byte,
+	blockVrfSig []byte,
 ) (SignedBlock, error) {
-	var block SignedBlock = &statelessBlock{
+	block := &statelessBlock{
 		StatelessBlock: statelessUnsignedBlock{
 			ParentID:     parentID,
 			Timestamp:    timestamp.Unix(),
 			PChainHeight: pChainHeight,
 			Certificate:  nil,
 			Block:        blockBytes,
+			VRFSig:       blockVrfSig,
 		},
 		timestamp: timestamp,
 	}
-
-	bytes, err := Codec.Marshal(CodecVersion, &block)
+	bytes, err := marshalBlock(block)
 	if err != nil {
 		return nil, err
 	}
 
 	return block, block.initialize(bytes)
 }
 
+func CalculateVRFOut(vrfSig []byte) []byte {
+	// build the hash of the following struct:
+	// +-------------------------+----------+------------+
+	// |  prefix :               | [8]byte  | "rng-derv" |
+	// +-------------------------+----------+------------+
+	// |  vrfSig :               | [48]byte |  48 bytes  |
+	// +-------------------------+----------+------------+
+	if len(vrfSig) != 48 {
+		return nil
+	}
+	buffer := make([]byte, 8+48)
+	copy(buffer, "rng-derv")
+	copy(buffer[8:], vrfSig[:])
+	outHash := hashing.Hash256(buffer)
```

Comment:
This is attempting to cast `buffer` into the `hashing.Hash256` type (`[32]byte`) and then slicing the result. This isn't actually computing a hash.